src/envoy/http/jwt_auth/jwt_authenticator.cc

```diff
@@ -62,6 +62,17 @@ void JwtAuthenticator::Verify(HeaderMap &headers,
   headers_ = &headers;
   callback_ = callback;
 
+  // Per the spec
+  // http://www.w3.org/TR/cors/#cross-origin-request-with-preflight-0, CORS
+  // pre-flight requests shouldn't include user credentials.
+  if (headers_->Method() &&
+      LowerCaseString(kOptionsHttpMethod) ==
```

Comment:
I think Envoy has fixed string for OPTIONS

https://github.com/envoyproxy/envoy/blob/master/source/common/http/headers.h#L176

Reply:
I prefer to do this check in Fitler.cc right inside the decodeHeaders().

Reply:
decodeHeaders() does not have tests. Placing the check in Verify() will enable tests. Plus, placing the check in Verify() is equivalent to placing the check in decodeHeaders().

Reply:
Done: changed to use the OPTIONS fixed string from Envoy.